### PR TITLE
Add rego for MS.AAD.3.8v1 - Managed device required for MFA registration

### LINKED ISSUE #324: AAD new ScubaGear check to secure MFA registration
# 💡 Summary #

Create a new ScubaGear policy check that corresponds to the new baseline policy protecting MFA registration #323.

## Implementation notes ##

- [ ] Create a new policy in the report that corresponds to the baseline policy.
- [ ] Implement the ScubaGear Rego code for the new policy.

### PR DESCRIPTION
**Pull request by crutchfield**
#  #

## 🗣 Description ##

Add rego for new MS.AAD.3.8v1 - Managed device required for MFA registration.

## 💭 Motivation and context ##

Required to implement new policy from secure baseline.
closes #324 

## 🧪 Testing ##

Updated rego unit test for Aad to include new policy.  

1) Unit test - Rego Aad
2) Invoke-Scuba for aad.  Expect warnign since we are not implementing CAP on tenant

## 📷 Screenshots (if appropriate) ##

image

image

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [X] All relevant type-of-change labels have been added.
- [X] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [X] Tests have been added and/or modified to cover the changes in this PR.
- [X] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: Testing/Unit/Rego/AAD/AADConfig_03_test.rego [hunk 1]
```diff
@@ -1520,18 +1520,177 @@ test_State_Incorrect_V3 if {
...
+    }
+
+    RuleOutput := [Result | Result = Output[_]; Result.PolicyId == PolicyId]
+
+    count(RuleOutput) == 1
+    RuleOutput[0].RequirementMet
+    contains(RuleOutput[0].ReportDetails, "conditional access policy(s) found that meet(s) all requirements:")
+}
+
+test_InCorrect_ReportOnly if {
```
Comment (Sloane4) [inline Testing/Unit/Rego/AAD/AADConfig_03_test.rego:1558]: should be Incorrect

File: Testing/Unit/Rego/AAD/AADConfig_03_test.rego [hunk 2]
```diff
@@ -1520,18 +1520,177 @@ test_State_Incorrect_V3 if {
...
+    }
+
+    RuleOutput := [Result | Result = Output[_]; Result.PolicyId == PolicyId]
+
+    count(RuleOutput) == 1
+    RuleOutput[0].RequirementMet
+    contains(RuleOutput[0].ReportDetails, "conditional access policy(s) found that meet(s) all requirements:")
+}
+
+test_InCorrect_EmptyGrantControls if {
```
Comment (Sloane4) [inline Testing/Unit/Rego/AAD/AADConfig_03_test.rego:1651]: Should be Incorrect

File: Testing/Unit/Rego/AAD/AADConfig_03_test.rego [hunk 3]
```diff
@@ -1520,18 +1520,177 @@ test_State_Incorrect_V3 if {
...
+    }
+
+    RuleOutput := [Result | Result = Output[_]; Result.PolicyId == PolicyId]
+
+    count(RuleOutput) == 1
+    not RuleOutput[0].RequirementMet
+    RuleOutput[0].ReportDetails == "0 conditional access policy(s) found that meet(s) all requirements. <a href='#caps'>View all CA policies</a>."
+}
+
+test_InCorrect_No_Policy if {
```
Comment (Sloane4) [inline Testing/Unit/Rego/AAD/AADConfig_03_test.rego:1682]: Should be Incorrect

**Review by Sloane4 — APPROVED** [review]

File: Testing/Unit/Rego/AAD/AADConfig_03_test.rego (hunk 1 above)
Comment (crutchfield) [inline Testing/Unit/Rego/AAD/AADConfig_03_test.rego:1558]:
```suggestion
test_Incorrect_ReportOnly if {
```

File: Testing/Unit/Rego/AAD/AADConfig_03_test.rego [hunk 4]
```diff
@@ -1520,18 +1520,177 @@ test_State_Incorrect_V3 if {
...
```
Comment (crutchfield) [inline Testing/Unit/Rego/AAD/AADConfig_03_test.rego:1682]:
```suggestion
test_Incorrect_No_Policy if {
```

**Comment by tkol2022** [comment]
### Initial Test Results
I completed round 1 of testing. I tried a non-compliant and a compliant conditional access policy and was able to generate Fail and Pass as expected.

The next round of testing will focus on user / group exclusions. See my email.

File: Testing/Unit/Rego/Teams/TeamsConfig_07_test.rego [hunk 5]
```diff
@@ -245,7 +245,7 @@ test_AllowEmailIntoChannel_Correct_V3_multi if {
     RuleOutput[0].ReportDetails == "N/A: Feature is unavailable in GCC environments"
 }
 
-test_AllowEmailIntoChannel_Correct_V4 if {
+test_AllowEmailIntoChannel_Correct_V3 if {
```
Comment (crutchfield) [inline Testing/Unit/Rego/Teams/TeamsConfig_07_test.rego:248]:
```suggestion
test_AllowEmailIntoChannel_Correct_V4 if {
```

**Comment by crutchfield** [comment]
> ### Initial Test Results
> I completed round 1 of testing. I tried a non-compliant and a compliant conditional access policy and was able to generate Fail and Pass as expected.
> 
> The next round of testing will focus on user / group exclusions. See my email.



> ### Initial Test Results
> I completed round 1 of testing. I tried a non-compliant and a compliant conditional access policy and was able to generate Fail and Pass as expected.
> 
> The next round of testing will focus on user / group exclusions. See my email.

@tkol2022 Updated 3.8 to include user/group exclude exemptions.  Also adde additional unit test to check this use case.

**Comment by tkol2022** [comment]
### Final Tests
I completed my round of functional testing by performing some scenarios with user exclusions to ensure those are handled as expected in the code. No defects.

**Review by tkol2022 — APPROVED** [review]

**Comment by tkol2022** [comment]
@nanda-katikaneni Ready for merge testing.

**Pull request merged**